homeassistant/components/permobil/binary_sensor.py

```diff
@@ -0,0 +1,81 @@
+"""Platform for binary sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from mypermobil import BATTERY_CHARGING
+
+from homeassistant import config_entries
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import MyPermobilCoordinator
+from .entity import PermobilEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class PermobilBinaryRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    is_on_fn: Callable[[Any], bool]
+    available_fn: Callable[[Any], bool]
+
+
+@dataclass(frozen=True)
+class PermobilBinarySensorEntityDescription(
+    BinarySensorEntityDescription, PermobilBinaryRequiredKeysMixin
+):
+    """Describes Permobil binary sensor entity."""
```

Comment:
You can avoid the mixin when using `kw_only=True`
 
```suggestion
@dataclass(frozen=True, kw_only=True)
class PermobilBinarySensorEntityDescription(BinarySensorEntityDescription):
    """Describes Permobil binary sensor entity."""

    is_on_fn: Callable[[Any], bool]
    available_fn: Callable[[Any], bool]
```